crates/sui-types/src/lib.rs

```diff
@@ -268,3 +268,31 @@ fn is_object_struct(
         }
     }
 }
+
+#[cfg(test)]
+mod tests {
```

Comment:
Can you add a test parsing a type with a type parameter, like `Coin`?

Reply:
done, and two more tests for a vector of coin of sui, and dynamic field